src/mempool_space/websocket.rs

```diff
@@ -0,0 +1,67 @@
+use super::api::{BlockEvent, MempoolSpaceWebSocketMessage, MempoolSpaceWebSocketRequestMessage};
+
+use anyhow::{anyhow, Ok};
+use async_stream::stream;
+use futures_util::stream::Stream;
+use futures_util::{SinkExt, StreamExt};
+use std::time::Duration;
+use tokio_tungstenite::connect_async_tls_with_config;
+use tokio_tungstenite::tungstenite::protocol::Message;
+use url::Url;
+
+pub async fn connect_and_publish_message(
```

Comment:
I think this function should just be `subscribe_to_blocks` and take the mempool.space `../api` endpoint. Later it will take an optional height and blockhash checkpoint.

Reply:
@LLFourn Great, I've renamed it and updated it to expect only the URL for now 😁